.github/workflows/trivy-scan.yml

```diff
@@ -0,0 +1,126 @@
+name: Trivy CVE Dependency Scanner
+
+on:
+  schedule:
+    - cron: '0 0 * * *'
+  workflow_dispatch:
+
+jobs:
+  scan-latest-release:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Get Latest Release Docker Image Sha
+        id: latest-sha
+        run: |
+          # Get the latest released docker image sha
+          curl -sL https://api.github.com/repos/vmware-tanzu/carvel-secretgen-controller/releases/latest | jq -r '.assets[].browser_download_url' | wget -i -
+
+          echo ::set-output name=image::$(yq eval '.spec.template.spec.containers[0].image' release.yml -N -j | jq 'select(. != null)' -r)
+          echo ::set-output name=tag::$(curl -sL https://api.github.com/repos/vmware-tanzu/carvel-secretgen-controller/releases/latest | jq -r '.tag_name')
+      - name: Install Trivy
+        run: |
+          brew install aquasecurity/trivy/trivy
+      - name: Run Trivy
+        run: |
+          trivy image ${{ steps.latest-sha.outputs.image }}
+          trivy image --format json --output trivy-results-image-latest.json ${{ steps.latest-sha.outputs.image }}
+      - name: Check for new Vulnerabilities
+        run: |
+          set -o pipefail
+
+          summary="Trivy scan has found new vulnerabilities in ${{steps.latest-sha.outputs.tag}} check https://github.com/${{github.repository}}/actions/runs/${{github.run_id}}"
+
+          vulnCount=$(jq '[ .Results[].Vulnerabilities ] | length' trivy-results-image-latest.json)
+          if [[ $vulnCount -eq 0 ]]; then
+            summary="Trivy Scan has not found any new Security Issues in ${{steps.latest-sha.outputs.tag}}"
+          fi
+
+          echo "SUMMARY=$summary" >> $GITHUB_ENV
+      - name: Send Slack Notification
+        if: success()
+        uses: slackapi/slack-github-action@v1.15.0
+        env:
+          SLACK_BOT_TOKEN: ${{ secrets.SLACK_BOT_TOKEN }}
+        with:
+          channel-id: C02MTA7F2G1
+          slack-message: "${{ env.SUMMARY }}"
+      - name: Send Failure notification
+        if: failure()
+        uses: slackapi/slack-github-action@v1.15.0
+        env:
+          SLACK_BOT_TOKEN: ${{ secrets.SLACK_BOT_TOKEN }}
+        with:
+          channel-id: G01FTP43JMQ
```

Comment:
I think the channel is not being found for some reason. I didn't investigate, I just noticed the other scanner has been failing https://github.com/vmware-tanzu/carvel-kapp-controller/runs/4554781469?check_suite_focus=true

Might be worth double checking the channel id. The bot should have permissions to post in all channels/

Reply:
`G01FTP43JMQ` looks the same, no?

Reply:
maybe because it's private?